models/base.py

```diff
@@ -430,12 +443,17 @@ def get_name_in_current_ctx(self) -> str:
     cur_scope_abs = cur_scope.get_abs_name_ctx_list()
     self_name_abs = self.get_abs_name_ctx_list()
     assert cur_scope_abs[0] is self_name_abs[0]  # same root
-    assert len(cur_scope_abs) >= len(self_name_abs)  # not implemented otherwise
     common_len = 0
-    while cur_scope_abs[common_len + 1] is self_name_abs[common_len + 1]:
+    max_common_len = min(len(cur_scope_abs), len(self_name_abs))
+    while common_len + 1 < max_common_len and cur_scope_abs[common_len + 1] is self_name_abs[common_len + 1]:
       common_len += 1
-    assert common_len == len(self_name_abs) - 2  # not implemented otherwise
-    return "base:" * (len(cur_scope_abs) - len(self_name_abs) + 1) + self.name
+    full_name = self_name_abs[-1].get_abs_name()
+    subs = full_name.split("/")
```

Comment:
Why so complicated? I think we should not do logic based on the strings (`split("/")`) when we already have it splitted beforehand (`self_name_abs`).